src/wallet/walletdb.cpp

```diff
@@ -832,6 +839,9 @@ DBErrors WalletBatch::LoadWallet(CWallet* pwallet)
                     // Set tx_corrupt back to false so that the error is only printed once (per corrupt tx)
                     wss.tx_corrupt = false;
                     result = DBErrors::CORRUPT;
+                } else if (wss.descriptor_corrupt) {
+                    pwallet->WalletLogPrintf("Error: Corrupt wallet descriptor found in wallet %s. This can be fixed by removing wallet from the settings so it's not loaded at startup.\n", pwallet->GetName());
+                    return DBErrors::CORRUPT;
```

Comment:
In 8c52bbe5923cb3d78bfd37e429d85b9727cf2dae "wallet: bugfix, load wallet with an unknown descriptor cause fatal error"

I think this should mention that the descriptor is either unknown or corrupt as this will occur for unknown future descriptors.

Also, I think it would be better to added a `DBErrors::UNKNOWN_DESCRIPTOR` so that the real error gets propagated to the user interface rather than directing them to find the log file.

Reply:
`DBErrors::TOO_NEW` might be a good fit here

Reply:
sounds good. Pushed.